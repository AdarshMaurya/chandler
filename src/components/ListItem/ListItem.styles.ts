import {PixelRatio, StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  notLastRow: {
    borderBottomColor: '#96A5B3',
    borderBottomWidth: 1 / PixelRatio.get(),
  },
  typeText: {
    marginLeft: 2,
    color: '#96A5B3',
  },
});
