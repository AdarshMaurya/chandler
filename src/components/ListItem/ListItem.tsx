import React, {PureComponent} from 'react';
import {Text, View} from 'react-native';

import {commonStyles} from '@theme';
import {styles} from './ListItem.styles';

interface IListItemProps {
  title: string;
  subtitle?: string;
  infoRight?: string;
  lastItem?: boolean;
}

export class ListItem extends PureComponent<IListItemProps> {
  public render() {
    const {title, infoRight, subtitle, lastItem} = this.props;

    return (
      <View style={[styles.row, !lastItem ? styles.notLastRow : null]}>
        <Text>{title}</Text>
        {subtitle && <Text style={styles.typeText}>{subtitle}</Text>}
        {infoRight && <View style={commonStyles.flex} />}
        {infoRight && <Text style={styles.typeText}>{infoRight}</Text>}
      </View>
    );
  }
}
