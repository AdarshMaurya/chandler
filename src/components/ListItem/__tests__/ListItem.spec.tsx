import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {ListItem} from '../ListItem';

describe('Components', () => {
  describe('ListItem', () => {
    it('should renders correctly with a title', () => {
      expect(toJson(shallow(<ListItem title="test" />))).toMatchSnapshot();
    });

    it('should renders correctly with a subtitle', () => {
      expect(
        toJson(shallow(<ListItem title="My title" subtitle="My subtitle" />))
      ).toMatchSnapshot();
    });

    it('should renders correctly with a subtitle and last', () => {
      expect(
        toJson(
          shallow(
            <ListItem title="My title" subtitle="My subtitle" lastItem={true} />
          )
        )
      ).toMatchSnapshot();
    });

    it('should renders correctly with an info right', () => {
      expect(
        toJson(shallow(<ListItem title="My title" infoRight="Info right" />))
      ).toMatchSnapshot();
    });
  });
});
