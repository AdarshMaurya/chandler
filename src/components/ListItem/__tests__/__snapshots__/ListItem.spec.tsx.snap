// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ListItem should renders correctly with a subtitle 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingVertical": 10,
      },
      Object {
        "borderBottomColor": "#96A5B3",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    My title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#96A5B3",
        "marginLeft": 2,
      }
    }
  >
    My subtitle
  </Text>
</Component>
`;

exports[`Components ListItem should renders correctly with a subtitle and last 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingVertical": 10,
      },
      null,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    My title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#96A5B3",
        "marginLeft": 2,
      }
    }
  >
    My subtitle
  </Text>
</Component>
`;

exports[`Components ListItem should renders correctly with a title 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingVertical": 10,
      },
      Object {
        "borderBottomColor": "#96A5B3",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test
  </Text>
</Component>
`;

exports[`Components ListItem should renders correctly with an info right 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingVertical": 10,
      },
      Object {
        "borderBottomColor": "#96A5B3",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    My title
  </Text>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#96A5B3",
        "marginLeft": 2,
      }
    }
  >
    Info right
  </Text>
</Component>
`;
